peerprep-fe/src/app/signup/page.tsx

```diff
@@ -0,0 +1,168 @@
+'use client';
+import { useState } from 'react';
+import { Button } from '@/components/ui/button';
+import { Checkbox } from '@/components/ui/checkbox';
+import { Input } from '@/components/ui/input';
+import { GithubIcon } from 'lucide-react';
+import { Alert, AlertDescription, AlertTitle } from '@/components/ui/alert';
+import { useRouter } from 'next/navigation';
+import Link from 'next/link';
+
+export default function SignUpPage() {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [agreeTerms, setAgreeTerms] = useState(false);
+  const [error, setError] = useState('');
+  const router = useRouter();
+
+  const handleSignUp = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setError('');
+    const apiEndpoint = 'http://localhost:3001/users';
+    // const type = 'user';
+    if (password !== confirmPassword) {
+      router.push('/signup');
+      setError('Passwords do not match');
+      return;
+    }
+    const result = await fetch(apiEndpoint, {
```

Comment:
We got axios installed so I think its better to use axios for the fetch. Axios allows us to create a new axios client and just access the client instead of using the API here